backend/src/test/java/com/karankumar/bookproject/backend/service/UserCreatedShelfServiceTest.java

```diff
@@ -170,4 +171,16 @@ void findOrCreate_throwsException_ifShelfNameIsNull() {
                 .isThrownBy(() -> userCreatedShelfService.findOrCreate(null));
     }
 
+    @Test
+    void shelfWithNameExists() {
+        User user = User.builder().build();
+        userCreatedShelfRepository.save(new UserCreatedShelf("test1", user));
+        userCreatedShelfRepository.save(new UserCreatedShelf("test2", user));
+
+//I'm not familiar with you db mock but these saved users do not appear to be within the test db
```

Comment:
Have a look at the `canSaveNonNullShelf()` method to find out how to save a shelf

Reply:
@knjk04 apologies for the delay! Last week wound up being being rather crazy fir me. I'll take a look at the ```canSaveNonNullShelf()``` method and be sure to reach out if I run into any trouble.

Reply:
That's fine, thanks